src/main/java/no/nav/pto/veilarbportefolje/tiltakshendelse/TiltakshendelseRepository.kt

```diff
@@ -71,24 +68,36 @@ class TiltakshendelseRepository(private val db: JdbcTemplate) {
         val sql = "SELECT * FROM tiltakshendelse"
 
         try {
-            return db!!.queryForList(sql).stream().map { rs: Map<String, Any> -> tiltakshendelseMapper(rs) }
+            return db.queryForList(sql).stream().map { rs: Map<String, Any> -> TiltakshendelseMapper.tiltakshendelseMapper(rs) }
                 .toList()
         } catch (e: Exception) {
             secureLog.error(e.message, e)
             throw RuntimeException(e)
         }
     }
 
-    companion object {
-        private fun tiltakshendelseMapper(rs: Map<String, Any>): Tiltakshendelse {
-            return Tiltakshendelse(
-                rs[TILTAKSHENDELSE.ID] as UUID?,
-                DateUtils.toLocalDateTimeOrNull(rs[TILTAKSHENDELSE.OPPRETTET] as Timestamp?),
-                rs[TILTAKSHENDELSE.TEKST] as String?,
-                rs[TILTAKSHENDELSE.LENKE] as String?,
-                Tiltakstype.valueOf((rs[TILTAKSHENDELSE.TILTAKSTYPE] as String?)!!),
-                Fnr.of(rs[TILTAKSHENDELSE.FNR] as String?)
-            )
+    /**
+     * "Eldste tiltakshendelse" er den hendelsen med eldst opprettet-dato,
+     * ikkje nødvendigvis den fyrste som vart lagra i databasen.
+     * */
+    fun hentEldsteTiltakshendelse(fnr: Fnr): Tiltakshendelse? {
+        val sql = """
```

Comment:
Generelt bra jobba 💪 🏆 
Jeg bare lurte på akkurat sql setningen om kanskje den er litt komplisert i forhold til lesbarhet og forståelse for folk som kanskje ikke er like drevne i sql. om det kanskje bare kunne vært:

```
SELECT id, fnr, opprettet, tekst, lenke, tiltakstype_kode, avsender, sist_endret
              FROM tiltakshendelse
            WHERE fnr = ?
            ORDER BY opprettet LIMIT 1
``` 

Du har nok gjort noen vurderinger her jeg ikke har :P jeg har f.eks ikke tenkt så mye på ytelse annet enn at det nok ikke er så mange per f.nr så ja, kun et spm/forslag :D hehe

Reply:
Eg likte din betre, så byttar til den. Trur det var noko sånn eg såg for meg før eg starta. 

Bak den eg hadde ligg det meir "googla ting på ein fredag ettermiddag" enn "har tatt gjennomtenkte og godt begrunna val" 😅